contrib/kube-prometheus/manifests/grafana/grafana-deployment.yaml

```diff
@@ -27,6 +27,8 @@ spec:
             secretKeyRef:
               name: grafana-credentials
               key: password
+        - name: GF_SERVER_ROOT_URL
```

Comment:
Hey @brancz! It's a valid question you're bringing up and difficult to answer. I guess it comes down to what you consider the basic way of accessing the Grafana UI. My argument is that it's analogous to Kibana in [the official Elasticsearch/Fluentd/Kibana add-on](https://github.com/kubernetes/kubernetes/tree/master/cluster/addons/fluentd-elasticsearch) to k8s, and here one has chosen to support `kubectl proxy` as the basic way of accessing it. To me it makes sense, as it works right out of the box, and it's better than the other out of the box method that I'm aware of, i.e. `kubectl port-forward` as `kubectl proxy`, unlike the former, connects to the service rather than individual pods and provides a more stable connection (`kubectl port-forward` often doesn't last long before it terminates).

Indeed I am using kube-prometheus in tandem with the official EFK add-on, so to me it makes total sense that Grafana (from the former) is configured in the same way wrt. access as Kibana (from the latter). If it's a good solution for Kibana it should also be so for Grafana.

If you don't agree with `kubectl proxy` being the default way of accessing Grafana, would you say the same for Kibana in the official EFK add-on? Should the latter have done it differently?